vendor/assets/javascripts/behaviors/nested_resources.js

```diff
@@ -135,9 +135,15 @@ Cmsify.NestedResource.prototype.deserializeFields = function(fields) {
 };
 
 Cmsify.NestedResource.prototype.setIsModified = function () {
-  if (tinyMCE.editors.some(function(editor) { return editor.isDirty() }) ||
-      JSON.stringify(this.serializeFields()) != JSON.stringify(this.lastData)) {
+  var tinyMCEDirty = false;
+  tinyMCE.editors.some(function(editor) {
+    if (editor.isDirty()) tinyMCEDirty = true;
+  }.bind(this));
+
+  if (JSON.stringify(this.serializeFields()) !== JSON.stringify(this.lastData)) {
     this.isModified = true;
+  } else {
+    this.isModified = tinyMCEDirty;
```

Comment:
Basically we were setting isModified to true and there was no way to set it back to false if the form was reverted to it's original state, I can actually probably do it with the old style, let me give it a shot.